chips/sam4l/src/gpio.rs

```diff
@@ -79,6 +80,8 @@ pub enum PeripheralFunction {
 const BASE_ADDRESS: usize = 0x400E1000;
 const SIZE: usize = 0x200;
 
+pub static INTERRUPT_COUNT: AtomicUsize = AtomicUsize::new(0);
```

Comment:
It's not obvious to me why this needs to be atomic? It's only accessed in `enable/disable_interrupt`, which is not currently reentrant safe (i.e. consider a pin calling disable in non-ISR and then an ISR enabling the same interrupt, current code can race incorrectly)